### PR TITLE
cleanup: Add dynamically derived array sizes to the API.

### PR DESCRIPTION
**Pull request by iphydf**
This change is [Reviewable](https://reviewable.io/reviews/TokTok/c-toxcore/2525)

### CODE REVIEW

File: toxav/toxav.h [hunk 1]
```diff
@@ -737,8 +737,13 @@ void toxav_callback_audio_receive_frame(ToxAV *av, toxav_audio_receive_frame_cb
...
  * @param vstride V chroma plane stride.
  */
-typedef void toxav_video_receive_frame_cb(ToxAV *av, uint32_t friend_number, uint16_t width, uint16_t height,
-        const uint8_t y[], const uint8_t u[], const uint8_t v[], int32_t ystride, int32_t ustride, int32_t vstride,
+typedef void toxav_video_receive_frame_cb(
+        ToxAV *av, uint32_t friend_number,
+        uint16_t width, uint16_t height,
+        const uint8_t y[/*! max(width, abs(ystride)) * height */],
+        const uint8_t u[/*! max(width/2, abs(ustride)) * (height/2) */],
+        const uint8_t v[/*! max(width/2, abs(ustride)) * (height/2) */],
```
Comment (nurupo) [inline toxav/toxav.h:745]: `ustride` for the size of `v[]` too?

**Review by nurupo — CHANGES_REQUESTED** [review]

File: toxav/toxav.h (hunk 1 above)
Comment (iphydf) [reply toxav/toxav.h]: Good catch, thanks!

**Comment by nurupo** [comment]
Ok, will refrain from looking at it for now since it's a draft (didn't notice at first).

**Comment by iphydf** [comment]
> Ok, will refrain from looking at it for now since it's a draft (didn't notice at first).

It's ready for review now.

**Review by nurupo — APPROVED** [review]

**Pull request merged**